fabric-object-builder-api-v1/src/main/java/net/fabricmc/fabric/mixin/object/builder/ModelPredicateProviderRegistryAccessor.java

```diff
@@ -0,0 +1,42 @@
+/*
+ * Copyright (c) 2016, 2017, 2018, 2019 FabricMC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package net.fabricmc.fabric.mixin.object.builder;
+
+import org.spongepowered.asm.mixin.Mixin;
+import org.spongepowered.asm.mixin.gen.Invoker;
+
+import net.minecraft.client.item.ModelPredicateProvider;
+import net.minecraft.client.item.ModelPredicateProviderRegistry;
+import net.minecraft.item.Item;
+import net.minecraft.util.Identifier;
+
+@Mixin(ModelPredicateProviderRegistry.class)
+public interface ModelPredicateProviderRegistryAccessor {
+	@Invoker
+	static ModelPredicateProvider callRegister(Identifier id, ModelPredicateProvider provider) {
+		throw new AssertionError("mixin dummy");
+	}
+
+	// mixin doesn't care about descriptor, must put two "register" accessors in different places
+	@Mixin(ModelPredicateProviderRegistry.class)
+	interface Specific {
```

Comment:
Aren't inner class mixins prohibited by mixin?

I'd try launching the game here

Reply:
Not if they're also mixins, they aren't.

Reply:
inner class mixins are like compile to class files, and if mixin can find the class file mixin should be able to handle it. I recall resource loader uses inner class based mixin as well, such as https://github.com/FabricMC/fabric/blob/304e48eb3a21b07a9e97e798e2efd981b28dca71/fabric-resource-loader-v0/src/main/java/net/fabricmc/fabric/mixin/resource/loader/MixinKeyedResourceReloadListener.java#L50

Reply:
It's really only local classes which it can't do. Anonymous classes and lambdas are both explicitly supported, whilst static classes are only handled as they effectively compile as if they were in a separate file anyway.

Reply:
Local classes can be like `Outer$1Local` names, so those classes theoretically can be supported. Proguard convert those local classes to `a$b` fyi.

Reply:
Mixin can certainly inject into them, the problem is having a local class in a mixin. Because being local it captures the (static) parent class you end up with a situation where the local class is then capturing a Mixin type which could of course necessitate classloading it. This could be worked around as it is for lambdas and anonymous classes but really isn't worth the effort given people shouldn't be making their Mixins bigger than necessary to start with.

Reply:
Might be easier to just give this another name, the api to the mod dev will be the same

Reply:
@modmuss50 It is impossible.
Mixin accessors are weird in that their methods' mapping to target accessed methods only respect method name and totally disregards method descriptors. As a result, both methods will have `"register"` as key in mixin refmap if they are to be in the same mixin class, and hence they cannot exist in the same mixin class due to refmap key clash and must be present in different classes. Renaming the methods does not resolve the refmap clash.

Reply:
Moved to outer class.